python/lsst/summit/utils/tmaUtils.py

```diff
@@ -0,0 +1,1346 @@
+        # things are unknown
+        if not self._isValid:
+            return TMAState.UNINITIALIZED
+
+        # if we're not in engineering mode, i.e. we're under normal CSC
+        # control, then if anything is in fault, we're in fault. If we're
+        # engineering then some axes will move when others are in fault
+        if not self.engineeringMode:
+            if any(self._axesInFault):
+                return TMAState.FAULT
+        else:
+            # we're in engineering mode, so return fault state if ALL are in
+            # fault
+            if all(self._axesInFault):
+                return TMAState.FAULT
+
+        # if all axes are off, the TMA is OFF
+        if all(self._axesOff):
+            return TMAState.OFF
+
+        # we know we're valid and at least some axes are not off, so see if
+        # we're in motion if no axes are moving, we're stopped
+        if not any(self._axesInMotion):
+            return TMAState.STOPPED
+
+        # now we know we're initialized, and that at least one axis is moving
+        # so check axes for motion and in position. If all axes are tracking
+        # and all are in position, we're tracking the sky
+        if (all(self._axesTRACKING) and all(self._axesInPosition)):
+            return TMAState.TRACKING
+
+        # we now know explicitly that not everything is in position, so we no
+        # longer need to check that. We do actually know that something is in
+        # motion, but confirm that's the case and return SLEWING
+        if (any(self._axesInMotion)):
+            return TMAState.SLEWING
+
+        # if we want to differentiate between MOVING_POINT_TO_POINT moves,
+        # JOGGING moves and regular slews, the logic in the step above needs to
+        # be changed and the new steps added here.
+
+        raise RuntimeError('State error: fell through the state sieve - rewrite your logic!')
+
+
+class TMAEventMaker:
+    # the topics which need logical combination to determine the overall mount
+    # state. Will need updating as new components are added to the system.
+
+    # relevant column: 'state'
+    _movingComponents = [
+        'lsst.sal.MTMount.logevent_azimuthMotionState',
+        'lsst.sal.MTMount.logevent_elevationMotionState',
+    ]
+
+    # relevant column: 'inPosition'
+    _inPositionComponents = [
+        'lsst.sal.MTMount.logevent_azimuthInPosition',
+        'lsst.sal.MTMount.logevent_elevationInPosition',
+    ]
+
+    # the components which, if in fault, put the TMA into fault
+    # relevant column: 'powerState'
+    _stateComponents = [
+        'lsst.sal.MTMount.logevent_azimuthSystemState',
+        'lsst.sal.MTMount.logevent_elevationSystemState',
+    ]
+
+    def __init__(self, client=None):
+        if client is not None:
+            self.client = client
+        else:
+            self.client = makeEfdClient()
+        self.log = logging.getLogger(__name__)
+        self._data = {}
+
+    @staticmethod
+    def isToday(dayObs):
```

Comment:
For sure, if this was a general purpose method for finding out if a given dayObs was the current day, I'd agree with that definition. But it's a `staticmethod` on the class, whose purpose (which I'll clarify by adding the docs now, as you say) is simply to find out if we're working live or on past (_i.e._ unchanging) data, so if this is somehow in the future then something has gone pretty badly wrong, and I just wanted to save myself in case I accidentally had an off-by-one error somewhere which had that in the future somehow.